tests/aws/services/apigateway/test_apigateway_http.py

```diff
@@ -0,0 +1,85 @@
+import json
+
+import pytest
+import requests
+
+from localstack.testing.pytest import markers
+from localstack.utils.sync import retry
+from tests.aws.services.apigateway.apigateway_fixtures import api_invoke_url
+
+
+@pytest.fixture
+def add_http_integration_transformers(snapshot):
+    key_value_transform = [
+        "accept-encoding",  # TODO: We add an extra space when adding this header
+        "content-length",
+        "date",
+        "domain",
+        "host",
+        "user-agent",
+        "x-amzn-apigateway-api-id",
+        "x-amzn-tls-cipher-suite",
+        "x-amzn-tls-version",
+        "x-amzn-trace-id",
+        "x-forwarded-port",
+        "x-forwarded-proto",
+    ]
+    for key in key_value_transform:
+        snapshot.add_transformer(snapshot.transform.key_value(key, reference_replacement=False))
```

Comment:
Oh!! That is a nice touch. It was indeed a bit confusing when reading the snapshot.

I just ran it again without the `value_replacement`. And it works for all of them. I guess I got confused at some point and added it. Still a bit unclear with what this param does!

Reply:
If you don't use `reference_replacement `, then you don't need to set the `value_replacement`. 

Next deep-dive, direction snapshots! 😄 you've been so good with them already, we kinda missed the subject 😅
What `reference_replacement=True` (the default) does is that for the whole test, it will look for transformed values (indicated with transformers) and keep a cache of them with a counter. If it encounters this value anywhere else in the snapshot (even in a not transformed value), it will replace it with the snapshot "key" and its index (the counter). I think showing an example would be a bit more explicit, we can have a look tomorrow if you'd like!

Trying to draft a small example:
```python
# register a transformer for the "value_one" key
snapshot.add_transformer(snapshot.transform.key_value("value_one"))

dict_1 = {
    "value_one": "random_value123",
    "value_two": "test",
    "value_three": "random_value123:moretext"
}
snapshot.match("resp1", dict_1)

dict_2 = {
    "value_one": "random_value456",
    "value_four": "another_value:random_value456",
    "parent": "random_value123",
}
snapshot.match("resp2", dict_2)
```

The resulting snapshots would be:
```json
{
    "resp1": {
        "value_one": "<value_one:1>",
        "value_two": "test",
        "value_three": "<value_one:1>:moretext"
    },
    "resp2": {
        "value_one": "<value_one:2>",
        "value_four": "another_value:<value_one:2>",
        "parent": "<value_one:1>",
    }
}
```
I'm not sure that's clearer, hope that helps a bit, but we will get into details whenever we can. 

Side note: `value_replacement` can be used also for esthetic reason, snapshot keys will be "kebab-cased" for every capital letter, so `restAPI` would become `rest-a-p-i`, so you could set a `value_replacement=rest-api` instead.